node/config/load.go

```diff
@@ -17,15 +18,40 @@ import (
 
 // FromFile loads config from a specified file overriding defaults specified in
 // the def parameter. If file does not exist or is empty defaults are assumed.
-func FromFile(path string, def interface{}) (interface{}, error) {
+func FromFile(path string, def interface{}, lc loadabilityCheck) (interface{}, error) {
+	// check for loadability
 	file, err := os.Open(path)
 	switch {
 	case os.IsNotExist(err):
+		if err := lc.configNotFound(); err != nil {
+			return nil, err
+		}
 		return def, nil
```

Comment:
There shouldn't be a separate "default". Instead, just have one `default() (interface{}, error)` function.

Reply:
(possibly attached to the config loading options)

Reply:
I'll take a look and do it if its a small change but I'd like to limit the scope of the PR to avoid improving the existing lotus config code.